File: src/utils/web/config/novelInfoConfig.ts
export interface DirectoryConfig {
  chapterId: string;
  /**
   * 编码方式
   * */
  encoding: string;
}

export interface InfoConfig {
  /**
   * 小说名
   * */
  name: string;
  /**
   * 作者
   * */
  author: string;
  /**
   * 最后更新时间
   * */
  lastUpdateTime: string;
  /**
   * 最新章节
   * */
  latestChapterId: string;
  /**
   * 编码方式
   * */
  encoding: string;
  /**
   * 小说图片
   * */
  image: string;
  /**
   * 描述
   * */
  desc: string;
}

/**
 * @author sushao
 * @since 0.4.0
 * @version 0.4.0
 * @description 章节列表的配置
 * */
export interface NovelInfoConfig {
  directory: DirectoryConfig;
  info: InfoConfig;
}

File: data/src/store/config/novel_info_config.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct NovelInfoConfig {
    pub(crate) info: InfoConfig,
    pub(crate) directory: DirectoryConfig,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct InfoConfig {
    pub(crate) name: String,
    pub(crate) author: String,
    #[serde(rename = "lastUpdateTime")]
    pub(crate) last_update_time: String,
    #[serde(rename = "latestChapterId")]
    pub(crate) latest_chapter_id: String,
    pub(crate) encoding: String,
    pub(crate) image: String,
    pub(crate) desc: String,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct DirectoryConfig {
    #[serde(rename = "chapterId")]
    pub(crate) chapter_id: String,
    pub(crate) encoding: String,
}
